### ISSUE TITLE
WooPay button disabled on One Page Checkout's form

### ISSUE THREAD

**Issue by senadir** [Automattic/woocommerce-payments#9237]
### Describe the bug
On a One Page Checkout product page, the second WooPay button (the one shown in the Checkout form) is disabled, other ECE buttons are not shown as well.

### To Reproduce

1. Install [One Page Checkout](https://woocommerce.com/products/woocommerce-one-page-checkout/)
2. Create a new product and mark it as One Page Checkout.
3. Go to product page, notice that the sub product buttons are shown, but the one in the form aren't.

### Screenshots
<img width="668" alt="image" src="https://github.com/user-attachments/assets/819a19b7-6934-4801-a354-00a3d0496847">

### Expected behavior
Both buttons should render.

Running Mac 14, Chrome latest, with WooPayments 8.0.1.

**Comment by senadir** [comment]
Note that it doesn't matter if cart already had products, or if I add the product or not.

**Comment by pierorocca** [comment]
We list this plugin as compatible so this would be regression.  https://woocommerce.com/document/woopay-merchant-documentation/#compatible-extensions

**Comment by pierorocca** [comment]
This does look like an edge case though as this is really a combined product and checkout page and the buttons can't be there twice. i.e. they can't be set to enabled for both Product and Checkout pages.

@nikkivias super interesting scenario here. How would you recommend we handle this? Merchant settings warning, documentation, default to checkout option?

**Comment by senadir** [comment]
BTW this is the default One Page Checkout plugin behavior, which we (Rubik) maintain. We're not sure how would this work with Checkout block.

**Comment by pierorocca** [comment]
It kind of resembles the Pay for Order flow to me. There's a list of products presented, in this case only 1, and then the ways you're able to pay. On our end and/or with Team Rubik help, we need to suppress the "Product Page" buttons. If I understand correctly, One Page Checkout is applied on a per product basis so we can't globally disable express checkout on product pages.

@bborman22 what's your perspective?

**Comment by bborman22** [comment]
> If I understand correctly, One Page Checkout is applied on a per product basis so we can't globally disable express checkout on product pages.

Yeah, this would be my thinking on it. We'd only want one set of buttons on the page and we'd want those to be with the rest of the "checkout" focused area in my opinion. So the short logic is if product has single page checkout applied, hide the product page buttons, but I'd have to get more familiar with the functionality to give a fuller reply.

**Comment by pierorocca** [comment]
@aheckler in the interim, could we address this via documentation and footnote that merchants should NOT enabled any express checkouts on the product page if any products are configured for One Page Checkout? This is a temporary limitation until we or he plugin can programmatically disable the buttons in the product details area.

**Comment by aheckler** [comment]
Done. https://woocommerce.com/document/woopay-merchant-documentation/#compatible-extensions

EDIT: I ended up reverting this update after a discussion with Piero. Our policy to date has been "Don't document bugs", and this is more of an issue with OPC anyway. (For example, other gateways that load express checkout buttons will have the same issue I would think.)

**Comment by pierorocca** [comment]
Thanks @aheckler . What about for merchants not using WooPay?

**Comment by pierorocca** [comment]
@senadir I did a bit of digging and this issue isn't limited to WooPayments. Any gateway placing a button on the PDP will have the same result. I don't fault the gateways, this is a plugin issue. 

Effectively, One Page checkout is combining a PDP and Checkout pages together to create a new type of page. A new page that's not a core page. That's up to the plugin IMO to orchestrate an experience that isn't creating unintended consquences.

Here I've enabled the Stripe and WooPayments gateways and set both to show PRBs on the PDP, Cart, and Checkout pages.

![image](https://github.com/user-attachments/assets/3bd9ff20-8fc6-4709-bef8-086dc6048172)

**Comment by pierorocca** [comment]
@alexflorisca @bborman22  fyi I think One Page Checkout is a missed use case for eliminating the excessive - OR - separators?
